app/Http/Controllers/ChangeNickController.php

```diff
@@ -2,17 +2,76 @@
 
 namespace OGame\Http\Controllers;
 
+use Illuminate\Http\Request;
+use Illuminate\Http\RedirectResponse;
 use Illuminate\View\View;
+use OGame\Services\PlayerService;
 
 class ChangeNickController extends OGameController
 {
     /**
-     * Shows the notes popup page
+     * Shows the username change popup page
      *
+     * @param PlayerService $player
      * @return View
      */
-    public function overlay(): View
+    public function overlay(PlayerService $player): View
     {
-        return view('ingame.changenick.overlay');
+        $canUpdateUsername = true;
+        if ($lastChange = $player->getLastUsernameChange()) {
+            $canUpdateUsername = $lastChange->addWeek()->isPast();
+        }
+
+        return view('ingame.changenick.overlay')->with([
+            'currentUsername' => $player->getUsername(false),
+            'canUpdateUsername' => $canUpdateUsername,
+        ]);
+    }
+
+    /**
+     * Rename the player username.
+     *
+     * @param PlayerService $player
+     * @return RedirectResponse
+     */
+    public function rename(Request $request, PlayerService $player): RedirectResponse
+    {
+        // Get form data
+        $name = $request->input('db_character');
+        $password = $request->input('db_character_password');
+
+        // Validate last username update
+        if (($lastChange = $player->getLastUsernameChange()) && !$lastChange->addWeek()->isPast()) {
+            return $this->errorResponse(__('You can only change your username once per week.'));
+        }
+
+        // Validate password
+        if (empty($password) || !$player->validatePassword($password)) {
+            return $this->errorResponse(__('Wrong password!'));
+        }
+
+        // Validate username
+        $validationResult = $player->isUsernameValid($name);
+        if (!$validationResult['valid']) {
+            return $this->errorResponse($validationResult['error']);
+        }
+
+        // Update username
+        $player->setUsername($name);
+        $player->save();
+
+        // Return success response
+        return redirect()->route('overview.index')->with('success', __('Player name changed! Please log in again!'));
```

Comment:
Since we are no longer requiring the user to login again after username change, can you update this message and remove the last sentence?

Currently it shows this in the UI when the user changes their name via this method:
<img width="336" alt="Screenshot 2024-08-23 at 10 33 17" src="https://github.com/user-attachments/assets/cad11b8d-65be-493f-bf52-838e8f9ea45a">

Reply:
Hi for sure, i will update the message. I had left it as its still up to date the message being throwed after a username change on the official game.